### ISSUE TITLE
Staging Review - Accessibility Feedback - Design System Team Banner Official Gov't Site Web Component

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#54057]
## VFS acceptance criteria
- [ ] Let Platform know when the **Must** feedback has been incorporated 
- [ ] Leave any comments for feedback you decide _not_ to take
- [ ] VFS team closes the ticket

## Feedback
- Practice areas will document their feedback following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback).

**Must:**

**Should:**
- Version 3.2.0 of the [USWDS component](https://designsystem.digital.gov/components/banner#changelog) introduced a few accessibility improvements that don't seem to have made it to our component. None of them are absolute must-haves, but it would be great to sync with this latest version.

**Consider:**
There are a few things about the "Here's how you know" button that bother me, although none of them are critical. It's better to be consistent with the rest of the government than to try to fix these just for VA sites, but as the DST contributes back to the USWDS, I would love to see some conversation around them:
   - Looking at the vertical click target size.
      - See the forthcoming [WCAG 2.2 SC 2.5.8](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html). I'd like to see at least 4px padding above the button at the top of the page, and/or have the touch target bigger.
      - Whether or not this passes the existing [WCAG 2.1 SC 2.5.5](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) is... debatable? Does the button actually count as inline content? I would say no but it's not totally clear.
   - The link-like styling of a button, which deviates from material honesty in a way that doesn't feel necessary.
   - The chevron being added via CSS with `:after`, which means it ends up living outside the focus indicator even though it behaves like part of the button (part of the clickable target).
   - The button text doesn't make sense without context (eg. screen reader user just tabbing through or bringing up a list of buttons).
   - The button text doesn't really communicate what action the button takes.
      - Especially with the link styling, I know the first time I encountered this I didn't notice the chevron and expected to go to another page.

Again, the bigger/more important objective is matching the rest of the government -- so no immediate action required. But I would love to see this component evolve within the USWDS.

## Platform directions
- Remove the governance-team label
- Update "Issue Title"
- Link to collab cycle Request epic
- Add your feedback
- Remove the CC-Request label (only if you created the ticket using the New Issue link from the Collaboration Cycle Reviews board)
- Add assignees based on collab cycle touchpoint
  - **Design Intent**: VFS designer, VFS PM (optional), yourself (optional)
  - **Midpoint Review**: VFS PM, yourself (optional)
  - **Staging Review QA Only**: VFS PM, yourself (optional)

**Comment by caw310** [comment]
This may carry over into the next sprint to incorporate the 'should' recommended in Staging review.

**Comment by jamigibbs** [comment]
@briandeconinck Thanks again for your thoughtful feedback! 

I reviewed the USWDS a11y updates for the [latest version release](https://designsystem.digital.gov/components/banner/#changelog) and I wanted to run one of those updates by you. Here is the update in question:

> Added aria-hidden attribute to the flag icon. The flag icon in the banner component is purely decorative and we can safely remove it from the screen reader readout. More information: [uswds#4925](https://github.com/uswds/uswds/pull/4925)

They have updated their image flag to the following (specifically the `aria-hidden="true"`):

```html
<img aria-hidden="true" class="usa-banner__header-flag" src="./img/us_flag_small.png" alt="">
```

If I recall, I did not add `aria-hidden="true"` in our implementation because my understanding was that, to tell assistive technology to ignore an image, we could just use a "blank alternative text" attribute like the following: `alt=""`

```html
<img
  role="presentation" 
  class="header-flag" 
  src="https://s3-us-gov-west-1.amazonaws.com/content.www.va.gov/img/tiny-usa-flag.png"
  alt="" />
```

When I do try and add `aria-hidden="true"` though, I'm getting an [Axe linter error](https://dequeuniversity.com/rules/axe/4.6/presentation-role-conflict?application=axe-linter) (the error is not entirely clear to me honestly):

![Screenshot 2023-03-01 at 1 22 55 PM](https://user-images.githubusercontent.com/872479/222243219-71e4fd17-3543-48eb-946e-aa816a9195f8.png)

Are you able to provide any additional guidance for this?

**Comment by jamigibbs** [comment]
Hi @briandeconinck! Just a gentle poke to check if you had a chance to review my question above about how to handle the image. I'm leaning towards just using what we had originally rather than what uswds has done:

```
<img
  aria-hidden="true"
  class="header-flag" 
  src="https://s3-us-gov-west-1.amazonaws.com/content.www.va.gov/img/tiny-usa-flag.png"
  alt="" />
```

**Comment by briandeconinck** [comment]
Hi @jamigibbs! Thank you for the poke, and for flagging this! I hadn't noticed that they used both empty `alt` and `aria-hidden`.

You're correct, empty `alt` is sufficient to hide the image from assistive technology. It looks like [some past versions of NVDA announced "blank" for empty alt](https://www.powermapper.com/tests/screen-readers/labelling/img-null-alt/) rather than just ignoring it, so that might be something adding `aria-hidden` fixes? But I can't test that to verify and I'm not able to think of any other circumstances where adding `aria-hidden` adds any value.

From their commit history it looks like USWDS went back and forth on it, but I'm not sure what their thinking was. 🤷 

As for the axe error, images with empty `alt` have an [implicit role of presentation](https://www.w3.org/TR/html-aam-1.0/#el-img-empty-alt) and as the axe warning notes, you're not supposed to have both that role and `aria-hidden`. If there was a benefit to having both `alt` and `aria-hidden` then I would say you could ignore the linter here, but since I'm coming up empty on situations where this makes sense I'd say the linter is right.

**tl;dr** not sure what USWDS' intention was and your instincts are correct. Just empty alt is fine.

**Comment by jamigibbs** [comment]
Thanks @briandeconinck! I've added the other updates to the component in the PR below.

**Comment by briandeconinck** [comment]
Those updates look good, please feel free to close when you're ready!
